### PR TITLE
[Fixes #18657323] Loads organization selects as needed

### PR DESCRIPTION
**Pull request by brvtl**

### CODE REVIEW

**Comment by dalibor** [comment]
This does not works. Confusing UI and javascript does not work. See the image: http://img7.imagebanana.com/img/jpu4akle/Selection_030.png

File: config/initializers/timeout.rb [hunk 1]
```diff
@@ -1,2 +1,4 @@
 # config/initializers/timeout.rb
-Rack::Timeout.timeout = 29  # 1 second less than heroku
\ No newline at end of file
+TIMEOUT_IN_SECONDS = Rails.env != 'development' ? 60 : 29 # 1 second less than heroku
```
Comment (glennr) [inline config/initializers/timeout.rb]: this is wrong.

Put back 
  Rack::Timeout.timeout = 29  # 1 second less than heroku

and in  config/environments/development.rb, put

  Rack::Timeout.timeout = 60 #or whatever

File: config/initializers/timeout.rb [hunk 2]
```diff
@@ -1,2 +1,4 @@
...
+
+Rack::Timeout.timeout = TIMEOUT_IN_SECONDS
```
Comment (glennr) [inline config/initializers/timeout.rb]: this constant is not needed. see above comment.

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -217,8 +217,7 @@ module ApplicationHelper
   end
 
   def funding_organizations_select
-    orgs = Organization.find(:all, :order => 'lower(name)')
-    orgs.map{|o| [o.display_name(100), o.id]}
+    orgs = Organization.find(:all, :order => 'lower(name)').map{ |o| [o.display_name(100), o.id] }
```
Comment (glennr) [inline app/helpers/application_helper.rb]: line too long

File: app/views/shared/outlays/_implementer_splits.html.haml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+- styling = ""
+- styling = "error" unless ff.object.errors.empty?
+- styling += " deleting" if ff.object.marked_for_destruction?
+
+%tr.overflow_fix.fields{:class => styling}
+  %td.implementer_search
+    = ff.hidden_field :data_response_id, :value => @response.id
+    = ff.hidden_field :updated_at, :value => Time.now
+    = ff.hidden_field :organization_mask, :value => ff.object.try(:organization_id) 
+    -# = ff.select :provider_mask, funding_organizations_select, {:prompt => true}, :class => 'name js_implementer_select js_combobox f_from'
```
Comment (glennr) [inline app/views/shared/outlays/_implementer_splits.html.haml]: dont commit commented lines

File: app/views/shared/outlays/_implementers.html.haml [hunk 5]
```diff
@@ -3,6 +3,13 @@
 - # Can't paginate form if there are errors therefore only
 - # show splits with errors if there are more than 50 errors
 
+:javascript
+  var selectBox = "#{escape_javascript(select_tag :activity_implementer_splits_attributes_provider_mask, options_for_select(funding_organizations_select),  :class => "name js_implementer_select js_combobox f_from" )}"
```
Comment (glennr) [inline app/views/shared/outlays/_implementers.html.haml]: line way too long.

File: app/views/shared/outlays/_implementers.html.haml [hunk 6]
```diff
@@ -31,25 +38,7 @@
...
-                = ff.hidden_field :updated_at, :value => Time.now
-                -# the importer intentionally returns nil implementers
-                -# which should fail on save
-                = ff.hidden_field :organization_mask, :value => ff.object.try(:organization).try(:id)
-                -# perf hack - non editable provider name
-                %span.ui-autocomplete= ff.object.try(:organization).try(:name) || "< n/a >"
-              %td= ff.text_field :spend, :class => 'js_spend'
-              %td= ff.text_field :budget, :class => 'js_budget'
-              %td= link_to_remove_fields "x", ff, {:class => 'delete_row totals_callback', :callback => 'updateTotalsValuesCallback'}
+          = render 'shared/outlays/implementer_splits', :ff => ff if (bulk_review || show_only_errors) || outlay.errors.empty?
```
Comment (glennr) [inline app/views/shared/outlays/_implementers.html.haml:37]: move "if (bulk_review || show_only_errors) || outlay.errors.empty?" to a helper - shortens the line

File: features/users/reporters/can_import_files.feature [hunk 7]
```diff
@@ -30,4 +30,4 @@ Feature: Reporter can import/export workplans
 
   Scenario: Reporter cannot import and save without reviewing
     Then I should not see "Bulk Import"
-    And I should not see "As the System Administrator, you can upload large files"
\ No newline at end of file
+    And I should not see "As the System Administrator, you can upload large files"
```
Comment (glennr) [inline features/users/reporters/can_import_files.feature]: lets not have a lot of cuke steps testing for things _not_ existing...

**Comment by glennr** [comment]
delete ('X') button on the Implementer listing is inconsistent with the nice red 'X' on the project.funder listing. please fix

**Comment by glennr** [comment]
Also, make it very obvious to the user they can now edit the implementer field.
- give the input field focus when edit is clicked or tabbed to.
- AFA the user is concerned, the org field should behave exactly like the amount fields

File: public/javascripts/application.js [hunk 8]
```diff
@@ -1446,9 +1446,43 @@ var activities_new = activities_create = activities_edit = activities_update = o
       $(document).find('.js_add_nested').trigger('click');
     }
     numericInputField(".js_implementer_spend, .js_implementer_budget");
+
+    $(".edit_implementer_row").live('click', function (e) {
```
Comment (glennr) [inline public/javascripts/application.js]: method too long

**Comment by glennr** [comment]
Good work
- pls fix the browser degradation when tabbing to multiple fields 
- pls fix the delete 'X' to match the rounded red X from the funder listing

File: app/helpers/application_helper.rb (hunk 3 above)
Comment (glennr) [inline app/helpers/application_helper.rb]: bump

File: app/views/admin/organizations/_form.html.haml [hunk 9]
```diff
@@ -3,10 +3,10 @@
...
-        = f.select :raw_type, options_for_select(Organization::ORGANIZATION_TYPES, f.object.raw_type), { :include_blank => true  }, :label => 'Type', :hint => 'Raw type of organization', :label_html => { :class => 'indented' }, :input_html => { :class => 'string required' }, :wrapper_html => { :class => 'input-wrapper' }, :tabindex => autotab
-        = f.select :currency, options_for_select(currency_options_for_select, f.object.currency), { :include_blank => true }, :label => 'Default currency', :hint => 'Select the default currency for your organization', :label_html => { :class => 'indented' }, :tabindex => autotab
-        = f.text_field :fosaid, :label => "Fosaid", :required => true, :hint => "Fosaid or organization", :label_html => { :class => 'indented' }, :wrapper_html => { :class => 'input-wrapper' }, :tabindex => autotab
+        = f.text_field :name, :label => "Name", :required => true, :hint => "What is the name of the organization?", :label_html => { :class => 'indented' }, :wrapper_html => { :class => 'input-wrapper' }, :size => 50
```
Comment (glennr) [inline app/views/admin/organizations/_form.html.haml:6]: try to add helpers / variables to reduce line length

Comment (brvtl) [inline app/views/admin/organizations/_form.html.haml:6]: It's not just a problem here, it's a problem with alot of the ui. Presenters pattern?

Comment (glennr) [inline app/views/admin/organizations/_form.html.haml:6]: We need to apply more strict practices to app/views/

Presenters? Maybe. But helpers are a good/easy place to start.

Note a lot of low-hanging fruit with non-DRY code e.g. label_html and wrapper_html

Comment (brvtl) [inline app/views/admin/organizations/_form.html.haml:6]: I don't know if this is the place to start refactoring all the views...
If we are going to do one - we might as well do all

Comment (glennr) [inline app/views/admin/organizations/_form.html.haml:6]: Dont Live with Broken Windows
Fix bad designs, wrong decisions, and poor code when you see them.

http://pragprog.com/the-pragmatic-programmer/extracts/tips

Comment (glennr) [inline app/views/admin/organizations/_form.html.haml:6]: Its taking more of our time discussing this than it is to simply write the 2 or 3 helpers that will fix this view...

**Pull request merged**